src/components/EditableStockListRow.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import * as Utilities from '../utils/utilities';
+import * as ApiCalls from '../utils/apiCalls';
+
+export class EditableStockListRow extends Component {
+  state = {};
+  handleClick = (name, id, symbol, shares) => {
+    if (name === 'delete') {
+      this.props.onDelete(id);
+    }
```

Comment:
Suggestion:
```
if (name == 'delete') {
  ...
} else if (name == 'save') {
  ...
} else if ...
  ...
}
```
is a bit clearer what's going on and also let's you put in a final `else` to catch anything else (which would be an error , probably).

Another way of doing this is to create an map of functions (untested):
```
const handlers = {
   'delete': (id, symbol, shares) => { this.props.onDelete(id); },
   'save': (id, symbol, shares) => { this.props.onSaveOrCancel(id, symbol, shares); }
   'cancel': (id, symbol, shares) => this.props.onSaveOrCancel(); }
};
handleClick = (name, id, symbol,s hares) => {
    handlers[name](id, symbol, shares);
}
```

Reply:
oh i like that, fixing in next push

Reply:
filed #31